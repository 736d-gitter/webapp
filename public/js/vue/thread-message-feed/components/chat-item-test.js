const mount = require('../../__test__/vuex-mount');
const momentTimezone = require('moment-timezone');
const { createSerializedMessageFixture } = require('../../__test__/fixture-helpers');
const { default: ChatItem } = require('./chat-item.vue');

describe('thread-message-feed chat-item', () => {
  momentTimezone.tz.setDefault('Europe/London');
  const message = createSerializedMessageFixture();
  const defaultProps = {
    message,
    useCompactStyles: false
  };

  describe('snapshot', () => {
    it('with default props', () => {
      const { wrapper } = mount(ChatItem, defaultProps);
      expect(wrapper.element).toMatchSnapshot();
    });
    it('showing item actions', () => {
      const { wrapper } = mount(ChatItem, { ...defaultProps, showItemActions: true });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('compact styles', () => {
      const { wrapper } = mount(ChatItem, { ...defaultProps, useCompactStyles: true });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('error', () => {
      const { wrapper } = mount(ChatItem, {
        ...defaultProps,
        message: { ...message, error: true }
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it('loading', () => {
      const { wrapper } = mount(ChatItem, {
        ...defaultProps,
        message: { ...message, loading: true }
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
