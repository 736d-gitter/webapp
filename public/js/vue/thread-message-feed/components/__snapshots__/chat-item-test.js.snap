// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed chat-item snapshot compact styles 1`] = `
<div
  class="chat-item burstStart compact"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <!---->
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;

exports[`thread-message-feed chat-item snapshot error 1`] = `
<div
  class="chat-item burstStart syncerror"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <!---->
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;

exports[`thread-message-feed chat-item snapshot highlighted - scrolls into view 1`] = `
<div
  class="chat-item burstStart chat-item__highlighted"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <!---->
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;

exports[`thread-message-feed chat-item snapshot loading 1`] = `
<div
  class="chat-item burstStart"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <loading-spinner-stub
          class="message-loading-icon"
        />
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;

exports[`thread-message-feed chat-item snapshot showing item actions 1`] = `
<div
  class="chat-item burstStart"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <div
      class="chat-item__actions"
    >
      <i
        class="chat-item__icon icon-check chat-item__icon--read"
      />
       
      <i
        class="chat-item__icon icon-ellipsis"
      />
    </div>
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <!---->
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;

exports[`thread-message-feed chat-item snapshot with default props 1`] = `
<div
  class="chat-item burstStart"
>
  <div
    class="chat-item__container"
  >
    <div
      class="chat-item__aside"
    >
      <div
        class="chat-item__avatar"
      >
        <avatar-stub
          size="s"
          user="[object Object]"
        />
      </div>
    </div>
     
    <!---->
     
    <div
      class="chat-item__content"
    >
      <div
        class="chat-item__details"
      >
        <div
          class="chat-item__from"
        >
          Tomas Vik
        </div>
         
        <div
          class="chat-item__username"
        >
          @viktomas_gitlab
        </div>
         
        <a
          class="chat-item__time"
          href="/abc/def?at=5d147ea84dad9dfbc522317a"
          title="May 18, 2016 3:48 AM"
        >
          May 18 2016 03:48
        </a>
         
        <!---->
      </div>
       
      <div
        class="chat-item__text"
      >
        <span
          class="mention"
          data-link-type="mention"
          data-screen-name="MadLittleMods"
        >
          @MadLittleMods
        </span>
          Example message using a bit of  
        <code>
          code
        </code>
         and 
        <strong>
          bold
        </strong>
         to show how 
        <em>
          markdown
        </em>
         is stored.
      </div>
    </div>
  </div>
</div>
`;
